src/rgw/rgw_s3_filter.h

```diff
@@ -27,6 +27,7 @@ struct rgw_s3_key_filter {
       encode(prefix_rule, bl);
       encode(suffix_rule, bl);
       encode(regex_rule, bl);
+      encode(negative_filter, bl);
```

Comment:
we should make sure that we are backward compatible with a mixed cluster that has older RGWs .
this means that we are at ver2 of the struct, but support down to ver1:
```cpp
ENCODE_START(2, 1, bl);
    encode(prefix_rule, bl);
    encode(suffix_rule, bl);
    encode(regex_rule, bl);
    encode(negative_filter, bl);
ENCODE_FINISH(bl);
```